server/grpc_service.go

```diff
@@ -1884,6 +1963,41 @@ func checkStream(streamCtx context.Context, cancel context.CancelFunc, done chan
 	<-done
 }
 
+func (s *GrpcServer) getGlobalTSOFromTSOServer(ctx context.Context) (pdpb.Timestamp, error) {
+	ok, forwardedHost, err := s.GetServicePrimaryAddr(ctx, "tso")
+	if !ok {
+		return pdpb.Timestamp{}, ErrNotFoundTSOAddr
+	}
+	if err != nil {
+		return pdpb.Timestamp{}, errors.WithStack(err)
+	}
+	client, err := s.getDelegateClient(ctx, forwardedHost)
+	if err != nil {
+		return pdpb.Timestamp{}, err
+	}
+	done := make(chan struct{})
+	ctx, cancel := context.WithTimeout(s.ctx, defaultTSOProxyTimeout)
+	go checkStream(ctx, cancel, done)
+	forwardStream, err := tsopb.NewTSOClient(client).Tso(ctx)
```

Comment:
Why do we need to create a stream here?

Reply:
I will cache it until forward-host changed